### PR TITLE
Code Review for <setup-associations> | Ruby School Library App

### PR DESCRIPTION
**Pull request by luigirazum**
# Hello Dear Code Reviewers Team 👋,
#### - 🎯 General requirements
- 📌 Make sure that there are no linter errors.
  - ✅ Linters for Ruby were installed. `Rubocop` reported no errors.
- 📌 Make sure that you used the correct GitFlow.
  - ✅ `setup-associations` feature branch was created from `dev` to follow the `git-flow`.
- 📌 Make sure that you documented your work in a professional way.
  - ✅ `README` file was updated with all the Project 3/7 details.
#### - 🎯 Ruby requirements
- 📌 Follow the list of best practices for Ruby.
  - ✅ Best practices for `Ruby` were followed.
#### - 🎯 Project requirements
Create the remaining <kbd>Classes</kbd> for our **Ruby School Library App** and create the <kbd>Associations</kbd> between them.
- 📌 Create a class <kbd>Classroom</kbd> with the following:
  - ✅ `@label` instance variable, should be initialized in the constructor.
  - ✅ Setter and getter for `@label` (remember about `attr_accessor`).
- 📌 Create the `has-many`/`belongs-to` relationship between <kbd>Classroom</kbd> and <kbd>Student</kbd>.
The following should be implemented:
  - ✅ Create the `has-many` side (a <kbd>Classroom</kbd> has many <kbd>Student</kbd>s).
  - ✅ Create the `belongs-to` side (a <kbd>Student</kbd> belongs to a <kbd>Classroom</kbd>).
  - ✅ Make sure that when adding a <kbd>Student</kbd> to a <kbd>Classroom</kbd> it also sets the <kbd>Classroom</kbd> for the <kbd>Student</kbd>.
  - ✅ Make sure that when setting the <kbd>Classroom</kbd> for a <kbd>Student</kbd> it also adds it to the <kbd>Classroom</kbd>s' <kbd>Student</kbd>s.
- 📌 Create a class <kbd>Book</kbd> with the following:
  - ✅ `@title` and `@author` instance variables, should be initialized in the constructor.
  - ✅ Setters and getters for instance variables (remember about `attr_accessor`).
- 📌 Create a class <kbd>Rental</kbd> with the following:
  - ✅ `@date` instance variable, should be initialized in the constructor.
  - ✅ Setter and getter for `@date` (remember about `attr_accessor`).
- 📌 Create the `many-to-many` (also `has-many-through`) relationship between <kbd>Person</kbd> and <kbd>Book</kbd> using the intermediate class <kbd>Rental</kbd>.
The following should be implemented:
  - ✅ Create the `has-many` side of <kbd>Book</kbd> and <kbd>Rental</kbd> (a <kbd>Book</kbd> has many <kbd>Rental</kbd>s).
  - ✅ Create the `belongs-to` side of <kbd>Rental</kbd> and <kbd>Book</kbd> (a <kbd>Rental</kbd> belongs to a <kbd>Book</kbd>).
  - ✅ Create the `has-many` side of <kbd>Person</kbd> and <kbd>Rental</kbd> (a <kbd>Person</kbd> has many <kbd>Rental</kbd>s).
  - ✅ Create the `belongs-to` side of <kbd>Rental</kbd> and <kbd>Person</kbd> (a <kbd>Rental</kbd> belongs to a <kbd>Person</kbd>).
  - ✅ Modify the constructor of <kbd>Rental</kbd> so <kbd>Book</kbd> and <kbd>Person</kbd> are set in it.
- 📌 See the <kbd>Associations</kbd> in action
    - ✅ A 'main.rb' was created on the 'root' folder, you can run it with `>ruby main.rb` to see how the associations work.

  ```ruby
  # a chemistry classroom is created
  chemistry = Classroom.new('chemistry')

  # a student is created, he is in chemistry classroom
  student = Student.new(22, chemistry, 'maximilianus')

  # a book1 is created, it hasn't been rented
  book1 = Book.new('chemistry for nubbies', 'ruth green')

  # a book2 is created, it hasn't been rented
  book2 = Book.new('chemistry advanced', 'john fitzgerald')

  # a teacher is created, with chemistry specialization
  teacher = Teacher.new(22, 'chemistry', 'mr. smith')

  # teacher rents book1 on may 25, 2023
  teacher.add_rental('05/25/2023', book1)

  # book2 is rented by student on may 25, 2023
  book2.add_rental('05/25/2023', student)
  ```

  The result of the previous code once you ran it should be:
  ```irb
  ❯ ruby main.rb
  ---- create a classroom ---
  classroom: chemistry
  classroom students: []
  
  ---- create a student ---
  student: maximilianus
  student classroom: chemistry
  student rentals: []

  ...
  ...
  ```

### Luis 🙋‍♂️

### CODE REVIEW

File: classes/person.rb [hunk 1]
```diff
@@ -4,15 +4,19 @@
 # There might be several variations of these classes.
 #  - 'correct_name' method is the operation implementation.
 class Person < Nameable
+  attr_reader :id, :rentals
   attr_accessor :name, :age
```
Comment (Deepakdanger) [inline classes/person.rb:8]: - [ ] There is no need of `id` in attr_reader & `rentals` will be included in attr_accessor.

File: classes/person.rb [hunk 2]
```diff
@@ -4,15 +4,19 @@
...
-  attr_reader :id
 
+  # Added associations
+  # a 'Person' <has-many> 'Rental's
```
Comment (Deepakdanger) [inline classes/person.rb:11]: - [ ] Here we should also write a method to `add_rental` here
```ruby
 def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    Rental.new(date, book, self)
  end
```

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @luigirazum , 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class Book
+  attr_reader :rentals
+  attr_accessor :title, :author
```
[the remark this answers is not in the document]
Comment (luigirazum) [reply classes/book.rb]: https://github.com/luigirazum/ruby-school-library-app/pull/3/commits/772e61e54da47f4fa3597ff82a8e0e01f6d8f719#diff-ee4e22a5d6ed72e9d0c08b0c1ed23e071d4beb1356e34e6def30ebee8a08a454R1-L3

[the remark this answers is not in the document]
Comment (luigirazum) [reply classes/book.rb]: All right! It's done!

File: classes/book.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+  # a 'Book' <has-many> 'Rental's
+  # is set with @rentals = []
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
[the remark this answers is not in the document]
Comment (luigirazum) [reply classes/book.rb:12]: Thanks for the feedback... the change is done!

File: classes/person.rb (hunk 1 above)
Comment (luigirazum) [reply classes/person.rb]: Done with that too!

File: classes/person.rb (hunk 2 above)
Comment (luigirazum) [reply classes/person.rb:11]: Fixed this one too!

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @luigirazum, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**